### ISSUE TITLE
text will not fit window until resize in awesomewm

### ISSUE THREAD

**Issue by thatarchguy** [arybczak/ncmpcpp#27]
![out](https://cloud.githubusercontent.com/assets/2035462/6122534/1c373806-b0c3-11e4-9084-a389a0cd5317.gif)

That gif should describe it better than I can with words.

**Comment by arybczak** [comment]
What ncmpcpp version is this?

**Comment by terraqout** [comment]
I have had  a same issue, but looks like in 0.6.2 it was fixed. Thanks @arybczak

**Comment by trygveaa** [comment]
I experience the problem in 0.6.2, though not all of the time, just once in a while.

**Comment by thatarchguy** [comment]
nuke@arch (~) $: ncmpcpp -v
ncmpcpp 0.6.2_pre

**Comment by arybczak** [comment]
Can you try the newest version?

**Comment by thatarchguy** [comment]
Because I'm just using the one in the official arch repos. I suppose I wrongly assumed it was the current version.

**Comment by arybczak** [comment]
Shit, it looks like I forgot to remove _pre suffix in 0.6.2. Sorry about my little rant then.

**Comment by thatarchguy** [comment]
it's been doing this for me ever since the .6 big update changing everything. I know this doesn't quite narrow it down though,

**Comment by terraqout** [comment]
@thatarchguy 
This is an issue of xterm (for me urxvt). Ncmpcpp as many other ncurses (pacmixer, etc.) draw themselves in measures of terminal geometry. Default is something near 80x40 old as holy crap standard.

In my case it was enough: 
```echo 'URxvt.geometry: 90x90' >> .Xresources```

or @arybczak will implement redraw signal to terminal in a first draw. But it would be a flickering effect on launch. Maybe just an option. 

Thanks, it should be closed soon

**Comment by arybczak** [comment]
Does applying this diff help by any chance?

```
diff --git a/src/ncmpcpp.cpp b/src/ncmpcpp.cpp
index 4bdf1f0..e6adc94 100644
--- a/src/ncmpcpp.cpp
+++ b/src/ncmpcpp.cpp
@@ -53,7 +53,7 @@ namespace
 {
 	std::ofstream errorlog;
 	std::streambuf *cerr_buffer;
-	bool run_resize_screen = false;
+	bool run_resize_screen = true;
 	
 	void sighandler(int sig)
 	{
```

**Comment by thatarchguy** [comment]
![out](https://cloud.githubusercontent.com/assets/2035462/7442168/0ea6bd6e-f0d4-11e4-8a6a-faa11f9c185f.gif)

Works! Thanks.

**Comment by arybczak** [comment]
Hmm, interesting. Can you also try this one (without the previous one)?

```
diff --git a/src/ncmpcpp.cpp b/src/ncmpcpp.cpp
index 4bdf1f0..4db39b9 100644
--- a/src/ncmpcpp.cpp
+++ b/src/ncmpcpp.cpp
@@ -103,6 +103,12 @@ int main(int argc, char **argv)
 	cerr_buffer = std::cerr.rdbuf();
 	std::cerr.rdbuf(errorlog.rdbuf());
 	
+	signal(SIGWINCH, sighandler);
+	// we get it after connection with mpd is broken.
+	// just ignore it and wait for the connection to
+	// be reestablished.
+	sigignore(SIGPIPE);
+
 	NC::initScreen(Config.colors_enabled);
 	
 	Actions::OriginalStatusbarVisibility = Config.statusbar_visibility;
@@ -150,12 +156,6 @@ int main(int argc, char **argv)
 	mouseinterval(0);
 	if (Config.mouse_support)
 		mousemask(ALL_MOUSE_EVENTS, 0);
-	
-	signal(SIGWINCH, sighandler);
-	// we get it after connection with mpd is broken.
-	// just ignore it and wait for the connection to
-	// be reestablished.
-	sigignore(SIGPIPE);
 
 	while (!Actions::ExitMainLoop)
 	{
```

**Comment by thatarchguy** [comment]
That works as well.

**Comment by terraqout** [comment]
@arybczak @thatarchguy  Thanks!

**Comment by arybczak** [comment]
@thatarchguy Ah cool, thanks for testing. The fix is in 0.6.4 and master.
